client/my-sites/marketplace/components/billing-interval-switcher/index.tsx

```diff
@@ -54,6 +54,23 @@ const BillingIntervalSwitcher: FunctionComponent< Props > = ( props: Props ) =>
 	const monthlyLabel = translate( 'Monthly' );
 	const annualLabel = translate( 'Annually' );
 
+	const searchParams = new URLSearchParams( document.location.search );
+	const monthlyParam = searchParams.get( 'monthly' );
+	const annualyParam = searchParams.get( 'annually' );
```

Comment:
Is it possible to utilize a general name for the query parameter? Something like `period` or `interval` rather than including `?monthly` or `?annually` in the URL, so that we can use `?period=monthly` or `?interval=annually`.

Reply:
`?interval=annually` feels like quite a neat solution to me, especially if we are able to provide mutli-year premium plugin subs as we do for plan subs (`bi-annually` + `triennially`.

Is there any specific reason aside from speed that makes using a flag a more attractive approach here?

Reply:
> Is there any specific reason aside from speed that makes using a flag a more attractive approach here?

No strong reason, if `?interval=annually` is a good solution I'll update the PR for this format, thanks for the input.

Reply:
Updated here https://github.com/Automattic/wp-calypso/pull/79552/commits/3d5cab87af883a9d029ab30d6f8653333f0322a9